pkg/sql/catalog/typedesc/virtual_table_record_type.go

```diff
@@ -0,0 +1,316 @@
+
+// GetName implements the NameKey interface.
+func (v VirtualTableRecordType) GetName() string { return v.desc.GetName() }
+
+// GetParentID implements the NameKey interface.
+func (v VirtualTableRecordType) GetParentID() descpb.ID { return v.desc.GetParentID() }
+
+// GetParentSchemaID implements the NameKey interface.
+func (v VirtualTableRecordType) GetParentSchemaID() descpb.ID { return v.desc.GetParentSchemaID() }
+
+// GetID implements the NameEntry interface.
+func (v VirtualTableRecordType) GetID() descpb.ID { return v.desc.GetID() }
+
+// IsUncommittedVersion implements the Descriptor interface.
+func (v VirtualTableRecordType) IsUncommittedVersion() bool { return v.desc.IsUncommittedVersion() }
+
+// GetVersion implements the Descriptor interface.
+func (v VirtualTableRecordType) GetVersion() descpb.DescriptorVersion { return v.desc.GetVersion() }
+
+// GetModificationTime implements the Descriptor interface.
+func (v VirtualTableRecordType) GetModificationTime() hlc.Timestamp {
+	return v.desc.GetModificationTime()
+}
+
+// GetDrainingNames implements the Descriptor interface.
+func (v VirtualTableRecordType) GetDrainingNames() []descpb.NameInfo {
+	// TODO(jordan): I don't think we need to return anything here, because this
+	// descriptor doesn't really "own" any names on top of what the table already
+	// does.
+	return nil
+}
+
+// GetPrivileges implements the Descriptor interface.
+func (v VirtualTableRecordType) GetPrivileges() *descpb.PrivilegeDescriptor {
+	return v.privs
+}
+
+// DescriptorType implements the Descriptor interface.
+func (v VirtualTableRecordType) DescriptorType() catalog.DescriptorType {
+	return catalog.Type
+}
+
+// GetAuditMode implements the Descriptor interface.
+func (v VirtualTableRecordType) GetAuditMode() descpb.TableDescriptor_AuditMode {
+	return descpb.TableDescriptor_DISABLED
+}
+
+// Public implements the Descriptor interface.
+func (v VirtualTableRecordType) Public() bool { return v.desc.Public() }
+
+// Adding implements the Descriptor interface.
+func (v VirtualTableRecordType) Adding() bool {
+	v.panicNotSupported("Adding")
+	return false
+}
+
+// Dropped implements the Descriptor interface.
+func (v VirtualTableRecordType) Dropped() bool {
+	v.panicNotSupported("Dropped")
+	return false
+}
+
+// Offline implements the Descriptor interface.
+func (v VirtualTableRecordType) Offline() bool {
+	v.panicNotSupported("Offline")
+	return false
+}
+
+// GetOfflineReason implements the Descriptor interface.
+func (v VirtualTableRecordType) GetOfflineReason() string {
+	v.panicNotSupported("GetOfflineReason")
+	return ""
+}
+
+// DescriptorProto implements the Descriptor interface.
+func (v VirtualTableRecordType) DescriptorProto() *descpb.Descriptor {
+	v.panicNotSupported("DescriptorProto")
+	return nil
+}
+
+// GetReferencedDescIDs implements the Descriptor interface.
+func (v VirtualTableRecordType) GetReferencedDescIDs() (catalog.DescriptorIDSet, error) {
+	return v.desc.GetReferencedDescIDs()
```

Comment:
I don't think this matters much yet, but this isn't accurate. Presently this will include sequence references, foreign key references, etc. Instead this should be `GetIDClosure()` supplemented with `v.GetID()`, `v.GetParentID()` and `v.GetParentSchemaID()`.